packages/evolution-common/src/services/baseObjects/Segment.ts

```diff
@@ -0,0 +1,511 @@
+/*
+ * Copyright 2024, Polytechnique Montreal and contributors
+ *
+ * This file is licensed under the MIT License.
+ * License text available at https://opensource.org/licenses/MIT
+ */
+
+import { Optional } from '../../types/Optional.type';
+import { IValidatable, ValidatebleAttributes } from './IValidatable';
+import { Uuidable, UuidableAttributes } from './Uuidable';
+import { WeightableAttributes, Weight, validateWeights } from './Weight';
+import * as SAttr from './attributeTypes/SegmentAttributes';
+import { Junction, JunctionAttributes } from './Junction';
+import { Routing, RoutingAttributes } from './Routing';
+import { Result, createErrors, createOk } from '../../types/Result.type';
+import { ParamsValidatorUtils } from '../../utils/ParamsValidatorUtils';
+import { ConstructorUtils } from '../../utils/ConstructorUtils';
+
+export const segmentAttributes = [
+    '_weights',
+    '_isValid',
+    '_uuid',
+    'modeCategory',
+    'mode',
+    'modeOtherSpecify',
+    'departureDate',
+    'arrivalDate',
+    'departureTime',
+    'arrivalTime',
+    'driver',
+    'driverUuid',
+    'carType',
+    'vehicleOccupancy'
+];
+
+export const segmentAttributesWithComposedAttributes = [
+    ...segmentAttributes,
+    'origin',
+    'destination',
+    'transitDeclaredRouting',
```

Comment:
I'm still unsure about having one field per mode declared here, there can be a variety, but if they are in composed attributes, that means they will be in a json data field anyway, so we can easily revisit this relationship later.

Reply:
See below. Composed attributes are standalone objects from other classes and will be saved in their own table in the db. Custom attributes are just values specific to a custom survey and will be saved as a json field in db.